HW06.Rmd

```diff
@@ -61,28 +91,58 @@ detach(package:EnsDb.Hsapiens.v75, unload = T)
 detach(package:ensembldb, unload = T)
 
 #add in your own gene of interest!!! 
-genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "CICP27", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2", "DDX11L1", "WBP1LP7", "SAMD11")
 
-
-GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest) %>% 
+# gather the data frame and tidy
+  gather(key = "cell_line", value = "TPM", 2:46) %>% 
+  separate(col = cell_line, into = c("1", "cell_line", "No."), sep = "\\.") %>% 
+  dplyr::select("gene_symbol", "cell_line", "TPM")
 
 #Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
 
-#print out the t-test results
+p <- c()
+i = 0
+for (x in genes_of_interest) {
+# filter data for certain gene
+  i = i + 1
+y <- GBM.genes.of.interest %>% 
+  filter(gene_symbol == x)
+#print out the t-test results as a table
+p[i] <- t.test(y$TPM[y$cell_line == "H3K27_Mutated"], y$TPM[y$cell_line == "WT"])$p.value
+}

```

Comment:
Good usage of the for loop to apply the t-test to all genes of interest and storing each result in a new vector (p[i]). After that it seems very simple to generate a new data frame that matches each gene with the corresponding t-test. Nice!